src/sentry/snuba/metrics/fields/base.py

```diff
@@ -658,16 +772,8 @@ def generate_bottom_up_dependency_tree_for_metrics(query_definition_fields_set):
     `CompositeEntityDerivedMetric` in a query definition fields set
     """
     dependency_list = []
-    for op, field_name in query_definition_fields_set:
-        if field_name not in DERIVED_METRICS:
-            # Instances of RawAggregatedMetric do not have dependencies
-            continue
-        derived_metric = DERIVED_METRICS[field_name]
-        # We are only interested in the dependency tree from instances of
-        # CompositeEntityDerivedMetric as they don't have a direct mapping to SnQL and so
-        # need to be computed post query which is practically when this function is called
-        if isinstance(derived_metric, CompositeEntityDerivedMetric):
-            dependency_list.extend(derived_metric.generate_bottom_up_derived_metrics_dependencies())
-        elif isinstance(derived_metric, SingularEntityDerivedMetric):
-            dependency_list.append((None, derived_metric.metric_name))
+    for op, metric_name in query_definition_fields_set:
+        dependency_list.extend(
+            metric_object_factory(op, metric_name).generate_bottom_up_derived_metrics_dependencies()
```

Comment:
I added this method to the FieldBase class so we don't have to do instance checks here. Also I had to make the lookup fo through metric_object_factory because otherwise the derived op was not detected here